app/shop_trip_functions.py

```diff
@@ -0,0 +1,127 @@
+# flake8: ignore=VNE001
+import json
+import math
+from datetime import datetime
+from pathlib import Path
+from typing import NamedTuple
+
+from app.customers.customer import Customer
+
+
+class Point(NamedTuple):
+    x: int  # noqa: VNE001 All my homies hate flake8!
+    y: int  # noqa: VNE001
+
+
+base_dir = Path(__file__).resolve().parent
+file_name = "config.json"
+
+
+def get_data_from_file() -> tuple:
+    with open(base_dir / file_name, "r") as file:
+        file_data = json.load(file)
+        return (
+            file_data["FUEL_PRICE"],
+            file_data["customers"],
+            file_data["shops"]
+        )
+
+
+def calculate_distance(p1: list, p2: list) -> float:
+    p1 = Point(*p1)
+    p2 = Point(*p2)
+
+    return math.sqrt((p2.x - p1.x) ** 2 + (p2.y - p1.y) ** 2)
+
+
+def calculate_fuel_spent(
+        fuel_price: float,
+        car: dict,
+        distance: float
+) -> float:
+    return car["fuel_consumption"] / 100 * distance * fuel_price
+
+
+class Cart:
+    def __init__(self, cart: dict, products: dict) -> None:
+        self.cart = cart
+        self.products = products
+
+    def calculate_product_cost(self) -> int | float:
+        total_cost = 0
+        for product in self.cart:
+            total_cost += self.cart[product] * self.products[product]
+        return total_cost
+
+    def generate_receipt(self) -> None:
+        for product in self.cart:
+            product_cost = self.cart[product] * self.products[product]
+
+            # removing .0 from product_cost
+            product_cost = int(product_cost) \
+                if product_cost % 1 == 0 else product_cost
+
+            print(
+                f"{self.cart[product]} {product}s for {product_cost} dollars"
+            )
```

Comment:
Let's move Point and Cart to the separate modules to have same structure throughout the code